app/components/user/create/username.js

```diff
@@ -0,0 +1,79 @@
+import AppComponent from '../../app';
```

Comment:
I would prefer that this is isolated into a filed component that is ready to pre-validate a field rather than a field just for username.

Reply:
In `user::create::username` component, I had done some extra work for checking username availability. That's why created a separate component to isolate this functionality and manage the state related to the username.